yt/units/__init__.py

```diff
@@ -117,6 +117,9 @@ def _value_updater(change):
 
 
 def _wrap_display_ytarray(arr):
-    from IPython.core.display import display
+    try:
+        from IPython.core.display import display
```

Comment:
This should be the exceptional path, not the first one we try.
Better yet would be to check on IPython's version at runtime instead of using try/except, so the code would self-document as obsolete when we drop support for older versions.